### PR TITLE
[Refactor] Optimize multi_count_distinct by CTE

### PR DESCRIPTION
**Pull request by Seaven**
## What type of PR is this：
- [ ] bug
- [ ] feature
- [x] enhancement
- [ ] others

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

1. rename RewriteMultiDistinctOptimizedRule to RewriteMultiDistinctWithoutGroupByRule

Reslove some questions:
1. `ForceCTE` is a global mark, it's will take other CTE which one need inline work error
2. Reslove count distinct function value, like: `select count(distinct x + 1), count(distinct y + 1) from t1`, the children of aggregate function not only column, maybe function
3. The origin optimize will split all aggregate function, we only need split count distinct function

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 82 / 83 (98.80%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/operator/OperatorBuilderFactory.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/Optimizer.java|2|2|100.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/RewriteMultiDistinctWithNoneGroupRule.java|74|74|100.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/RuleType.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/CTEContext.java|3|3|100.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/RewriteMultiDistinctRule.java|2|2|100.00%|

**Comment by Seaven** [comment]
run starrocks_fe_unittest

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/RewriteMultiDistinctWithNoneGroupRule.java [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+ *                   /           \
+ *           CTEProduce           Cross join
+ *             /                 /          \
+ *      Child Plan        Cross join       Agg[sum(v4), max(v5)]
+ *                        /          \            \
+ *                 Agg[cd(v1)]     Agg[cd(v2)]   CTEConsume
+ *                     /               \
+ *                CTEConsume        CTEConsume
+ */
+public class RewriteMultiDistinctWithNoneGroupRule extends TransformationRule {
```
Comment (kangkaisen) [inline fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/RewriteMultiDistinctWithNoneGroupRule.java:55]:
```suggestion
public class RewriteMultiDistinctWithoutGroupByRule extends TransformationRule {
```

**Review by kangkaisen — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 83 / 84 (98.81%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/operator/OperatorBuilderFactory.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/Optimizer.java|3|3|100.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/RewriteMultiDistinctWithoutGroupByRule.java|74|74|100.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/RuleType.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/CTEContext.java|3|3|100.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/RewriteMultiDistinctRule.java|2|2|100.00%|

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/CTEUtils.java [hunk 2]
```diff
@@ -144,15 +142,6 @@ private static void collectCteConsume(Group root, OptimizerContext context, bool
             if (collectCosts && !expression.getInputs().isEmpty()) {
                 context.getCteContext().addCTEConsumeInlineCost(consume.getCteId(),
                         calculateStatistics(expression.getInputs().get(0), context).getComputeSize());
-            } else if (collectCosts && expression.getInputs().isEmpty()) {
```
Comment (Youngwb) [inline fe/fe-core/src/main/java/com/starrocks/sql/optimizer/CTEUtils.java:147]: Why remove this

Comment (Seaven) [reply fe/fe-core/src/main/java/com/starrocks/sql/optimizer/CTEUtils.java:147]: Unuse, will enforce use CTE, don't need mock  consume costs

**Review by Youngwb — APPROVED** [review]

**Pull request merged**